db_access.py

```diff
@@ -24,16 +30,25 @@ def get_all_users():
     return conn.execute('SELECT id, handle, admin FROM user').fetchall()
 
 
+def user_for(uid):
+    conn = get_db()
+    for a in conn.execute('SELECT handle FROM user WHERE id=:id', {'id':uid}):
+        return a[0]
+    return ''
+
 def delete_user(user_id):
     conn = get_db()
     conn.execute('DELETE FROM user WHERE id = :id', {'id': user_id})
     conn.commit()
 
 
-def add_user(handle):
+def add_user(handle, password):
+    password = hash_ps(password)
     conn = get_db()
-    conn.execute('INSERT INTO user (handle, admin) values (:handle, :admin)',
-                 {'handle': handle, 'admin': 0})
+    for c in conn.execute('SELECT * FROM user WHERE handle=:handle', {'handle':handle}):
+        return list(c)
+    conn.execute('INSERT INTO user (handle, password, admin) values (:handle, :password, :admin)',
+                 {'handle': handle, 'password':password, 'admin': 0})
```

Comment:
You could just call the `hash_ps` method in the dictionary:

      {'handle': handle, 'password':hash_ps(password), 'admin': 0}

But if you would rather call it on it's own line, please name the variable something else, like `hashed_ps`